tap_google_ads/streams.py

```diff
@@ -238,11 +238,20 @@ def format_field_names(self):
                 self.stream_schema["properties"].pop("ad")
 
             if (
-                resource_name not in {"metrics", "segments"}
+                resource_name not in {"metrics", "segments", "user_interest"}
                 and resource_name not in self.google_ads_resource_names
             ):
                 self.stream_schema["properties"][resource_name + "_id"] = schema["properties"]["id"]
 
+            # user_interest stream has a `user_interest.user_interest_id` instead of a `user_interest.id`
+            # this case is handled below where it's set to id for the user_interest stream
+            # but remains user_interest_id when its an attributed resource
+            if (
+                resource_name in self.google_ads_resource_names
+                and resource_name == "user_interest"
+            ):
+                self.stream_schema["properties"]["id"] = schema["properties"]["user_interest_id"]
```

Comment:
Minor code ⛳ , but if we did something like this:

```
# user_interest stream has a `user_interest.user_interest_id` instead of a `user_interest.id`
# this case is handled below where it's set to id for the user_interest stream
# but remains user_interest_id when its an attributed resource
if resource_name == "user_interest":
    if resource_name in self.google_ads_resource_names:
        self.stream_schema["properties"]["id"] = schema["properties"]["user_interest_id"]
elif(
    resource_name not in {"metrics", "segments"}
    and resource_name not in self.google_ads_resource_names
):
    self.stream_schema["properties"][resource_name + "_id"] = schema["properties"]["id"]
```

We could consolidate all the user_interest actions in one block, instead of spanning it across two. Since it's a pretty specific implementation detail, having it in one place could potentially be nice. And make it easier to abstract away if we needed to in the future. The trade off there is a nested `if`.

Reply:
Irrelevant with new Class approach